jadx-gui/src/main/java/jadx/gui/ui/codearea/ClassCodeContentPanel.java

```diff
@@ -0,0 +1,74 @@
+package jadx.gui.ui.codearea;
+
+import java.awt.BorderLayout;
+
+import javax.swing.JTabbedPane;
+
+import jadx.gui.treemodel.JNode;
+import jadx.gui.ui.ContentPanel;
+import jadx.gui.ui.TabbedPane;
+import jadx.gui.utils.NLS;
+
+/**
+ * Displays one class with two different view:
+ *
+ * <ul>
+ * <li>Java source code of the selected class (default)</li>
+ * <li>Smali source code of the selected class</li>
+ * </ul>
+ */
+public final class ClassCodeContentPanel extends AbstractCodeContentPanel {
+	private static final long serialVersionUID = -7229931102504634591L;
+
+	private final CodePanel javaCodePanel;
+	private final CodePanel smaliCodePanel;
+	private JTabbedPane areaTabbedPane = new JTabbedPane(JTabbedPane.BOTTOM);
+
+	public ClassCodeContentPanel(TabbedPane panel, JNode jnode) {
+		super(panel, jnode);
+
+		javaCodePanel = new CodePanel(this);
+		smaliCodePanel = new CodePanel(this) {
+			@Override
+			protected AbstractCodeArea createCodeArea(ContentPanel contentPanel) {
+				return new SmaliArea(contentPanel);
+			}
+		};
```

Comment:
Can we replace this method override with additional parameter in `CodePanel` constructor?